File: backend-go/internal/dto/request/initiative.go
package request

type CreateInitiativeReq struct {
	WorkspaceID uint64   `json:"workspace_id"`
	Name        string   `json:"name" binding:"required"`
	Description string   `json:"description,omitempty"`
	Color       string   `json:"color,omitempty"`
	Status      string   `json:"status,omitempty"`
	TargetDate  string   `json:"target_date,omitempty"`
	StartDate   string   `json:"start_date,omitempty"`
	ProjectIDs  []uint64 `json:"project_ids,omitempty"`
}

type UpdateInitiativeReq struct {
	Name        string   `json:"name,omitempty"`
	Description *string  `json:"description,omitempty"`
	Color       *string  `json:"color,omitempty"`
	Status      string   `json:"status,omitempty"`
	TargetDate  *string  `json:"target_date,omitempty"`
	StartDate   *string  `json:"start_date,omitempty"`
	ProjectIDs  []uint64 `json:"project_ids,omitempty"`
}

File: frontend/src/types/initiative.ts
export interface Initiative {
  id: number
  workspace_id: number
  name: string
  description?: string
  color?: string
  status: string
  target_date?: string
  start_date?: string
  sort_order: number
  projects?: any[]
  created_by?: any
  created_at: string
}

export interface InitiativeCreateRequest {
  name: string
  description?: string
  color?: string
  status?: string
  target_date?: string
  start_date?: string
  project_ids?: number[]
}

export interface InitiativeUpdateRequest {
  name?: string
  description?: string
  color?: string
  status?: string
  target_date?: string
  start_date?: string
  project_ids?: number[]
}

export interface InitiativeProgress {
  total_issues: number
  completed_issues: number
  progress: number
  project_count: number
}
